### PR TITLE
feat(overlay): add getConversations, selectConversation, createConversation methods to overlay

### PR DESCRIPTION
**Pull request by mikitabut**
**Description:**

During necessity implemented next methods for overlay:
1. getConversations
2. selectConversation
3. createConversation

Also improved typing to have common types in chat and overlay
Also improved sandbox case 1 to reflect new changes 

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [ ] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [ ] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2228.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2228.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2228.nightly-test.deltixhub.io
>E2E tests status: success

**Review by irinakartun — DISMISSED** [review]

File: libs/shared/src/types/chat.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+export enum PublishActions {
+  ADD = 'ADD',
+  DELETE = 'DELETE',
+}
+
+export interface EntityPublicationInfo {
+  action?: PublishActions;
+  isNotExist?: boolean;
+  version?: string;
+  currentlySelectedVersion?: boolean;
```
Comment (Alexander-Kezik) [inline libs/shared/src/types/chat.ts]: could you please remove `currentlySelectedVersion` field?

Comment (mikitabut) [reply libs/shared/src/types/chat.ts]: Done

**Comment by mikitabut** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2228.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Review by irinakartun — APPROVED** [review]

**Pull request merged**